src/Cryptol/TypeCheck/SimpType.hs

```diff
@@ -262,6 +262,12 @@ tMax x y
   maxK (Nat 0) t = t
   maxK (Nat k) t
 
+    -- max 1 t ~> t,   if t = a ^ b && a >= 1
```

Comment:
Does `a >= 1` need to be in the constraint? Doesn't `max 1 t ~> t` also when `a == 0`?

Reply:
It doesn't work for `0`.  Consider, for example, `max 1 (0 ^ 2)`, this should result in `1`, and not `0 ^ 2`, which is `0`

Reply:
I guess I was thinking that `a` can equal `0` when `b == 0`. However, I can see it's difficult to get the boundary precise.